spec/rubocop/cop/style/if_unless_modifier_spec.rb

```diff
@@ -4,34 +4,37 @@
 # Note: most of these tests probably belong in the shared context "condition modifier cop"
 ######
 RSpec.describe RuboCop::Cop::Style::IfUnlessModifier, :config do
-  let(:ignore_cop_directives) { true }
-  let(:allow_uri) { true }
-  let(:line_length_config) do
-    {
-      'Enabled' => true,
-      'Max' => 80,
-      'AllowURI' => allow_uri,
-      'IgnoreCopDirectives' => ignore_cop_directives,
-      'URISchemes' => %w[http https]
-    }
-  end
-  let(:other_cops) { { 'Layout/LineLength' => line_length_config } }
+  shared_context 'with LineLength settings' do |enabled: true,
+                                                allow_uri: true,
+                                                ignore_cop_directives: true|
+    let(:other_cops) do
+      {
+        'Layout/LineLength' => {
+          'Enabled' => enabled,
+          'Max' => 80,
+          'AllowURI' => allow_uri,
+          'IgnoreCopDirectives' => ignore_cop_directives,
+          'URISchemes' => %w[http https]
+        }
+      }
+    end
+  end
+
+  include_context 'with LineLength settings' # default ones
 
   extra = ' Another good alternative is the usage of control flow `&&`/`||`.'
   it_behaves_like 'condition modifier cop', :if, extra
   it_behaves_like 'condition modifier cop', :unless, extra
 
-  context 'modifier if that does not fit on one line' do
-    let(:spaces) { ' ' * 59 }
-    let(:body) { "puts '#{spaces}'" }
-    let(:source) { "#{body} if condition" }
+  context 'modifier `if` that does not fit on one line' do
+    let(:body) { "puts '                                                           '" }
```

Comment:
Ha! I was trying to avoid defining methods to replace `let`s because of an argument that it's basically the same thing. 

But this `spaces` method clearly shows how methods can be better, since they do accept parameters. I really like it.

It's possible to go even further with something like:
```ruby
def body(spaces:)
  "puts '#{' ' * spaces}"
end
```

Reply:
I was trying to figure out a good way to factor out `body` but wasn't able to come up with an alternative that I thought I could sell. Personally, I'd probably actually copy paste it into each test. I think some duplication can be advantageous across tests, since it makes it a lot easier to write varying test cases, and it makes each test more self-contained. That, to me, is easier to maintain and understand without scrolling up and down the file (and other files) to trace through all of the implicit dependencies.

I think our programmer's inclination to DRY up code actually works against us when it comes to tests. In tests we often want logic that changes in small ways across each test, and that makes it harder to deduplicate without making things more difficult to understand and maintain.

Reply:
One sort of thing I like to think about is how hard it would be to rearrange the tests in a file. I often want to restructure the tests in a file to tell a more clear story. If the tests resist being moved around, then they're too coupled to various implicit dependencies. Thinking about rearranging the tests in this file gives me panic sweats.

Reply:
> Thinking about rearranging the tests in this file gives me panic sweats.

Indeed. And it's by far not the worst example.